src/Core/Kernel.php

```diff
@@ -365,15 +365,7 @@ protected function initializeDatabaseConnectionVariables(): void
 
         try {
             $setSessionVariables = (bool) EnvironmentHelper::getVariable('SQL_SET_DEFAULT_SESSION_VARIABLES', true);
-            $connectionVariables = [];
-
-            /**
-             * @deprecated tag:v6.7.0 - remove if clause and enforce timezone setting
-             */
-            $timeZoneSupportEnabled = (bool) EnvironmentHelper::getVariable('SHOPWARE_DBAL_TIMEZONE_SUPPORT_ENABLED', Feature::isActive('v6.7.0.0'));
-            if ($timeZoneSupportEnabled && $setSessionVariables) {
-                $connectionVariables[] = 'SET @@session.time_zone = "+00:00"';
-            }
+            $connectionVariables = ['SET @@session.time_zone = "+00:00"'];
```

Comment:
With this change now we basically circumvented the benefit of the `SQL_SET_DEFAULT_SESSION_VARIABLES` variable, as the whole purpose of that variable is to not execute a query this early in the stack

When we always do a query we can also always set the other variables, but i guess that is not what we want 🙈 

So i would think this should go into the if branch as well

However then at least for me locally the test always failed

Maybe we could instead extract this into a helper class (e.g. the existing MySQLFactory), then we could test it independendly and actually call the code we wanna test 🤔

Reply:
BTW before you also only set the variable when SQL_SET_DEFAULT_SESSION_VARIABLES was set to true